### PR TITLE
Add playhead

### PR DESCRIPTION
**Pull request by brandonreid**
Playhead ready! There are still issues with running that we'll want to investigate once pending tasks are removed from view:

https://github.com/PrefectHQ/graphs/assets/6776415/74c8136e-a7f9-43d6-8d6e-bfb301d5894e

After refresh:
image

### CODE REVIEW

File: src/factories/playhead.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+    if (
+      !viewport.moving
+      && playheadStartedVisible
+      && element.position.x > application.screen.width - autoViewportUpdatePadding
+    ) {
+      const originalLeft = scale.invert(viewport.left)
+
+      viewport.zoomPercent(-0.1, true)
+      viewport.left = scale(originalLeft)
```
Comment (pleek91) [inline src/factories/playhead.ts:53]: aren't these to lines just setting `viewport.left = viewport.left`? 

```
const originalLeft = scale.invert(viewport.left)
viewport.left = scale(originalLeft)
```

File: src/factories/playhead.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  const ticker = (): void => {
+    render()
+    checkViewport()
+  }
+
+  function render(): void {
+    playhead.width = config.styles.playheadWidth
+    playhead.height = application.stage.height
+    playhead.tint = config.styles.playheadColor
+    element.position.x = scale(new Date()) * viewport.scale._x + viewport.worldTransform.tx
```
Comment (pleek91) [inline src/factories/playhead.ts:34]: why `viewport.scale._x` and not `viewport.scale.x`?

File: src/factories/playhead.ts (hunk 1 above)
Comment (pleek91) [reply src/factories/playhead.ts]: Yeah seems like its the same as this
```
const originalLeft = viewport.left
viewport.zoomPercent(-0.1, true)
viewport.left = originalLeft
```

**Comment by pleek91** [comment]
@brandonreid pushed a commit to simplify this a bit. Double check that this still functions as it should. From my testing it works the same.

File: src/factories/playhead.ts (hunk 2 above)
Comment (brandonreid) [reply src/factories/playhead.ts]: Not sure tbh, grabbed from v1 but `.x` seems to work just fine so updated it.

**Comment by brandonreid** [comment]
> @brandonreid pushed a commit to simplify this a bit. Double check that this still functions as it should. From my testing it works the same.

LGTM! Tried it out, functions well. Nice cleanup!

**Review by pleek91 — APPROVED** [review]

**Pull request merged**